clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -922,20 +923,29 @@ class GCEUtil {
     String targetProxyType = Utils.getTargetProxyType(target)
     String targetProxyName = getLocalName(target)
 
-    def retrievedTargetProxy = null
-
+    def proxyGet = null
     switch (targetProxyType) {
       case "targetHttpProxies":
-        retrievedTargetProxy = compute.targetHttpProxies().get(project, targetProxyName).execute()
+        proxyGet = { compute.targetHttpProxies().get(project, targetProxyName).execute() }
         break
       case "targetHttpsProxies":
-        retrievedTargetProxy = compute.targetHttpsProxies().get(project, targetProxyName).execute()
+        proxyGet = { compute.targetHttpsProxies().get(project, targetProxyName).execute() }
         break
       default:
         log.warn("Unexpected target proxy type for $targetProxyName.")
-        retrievedTargetProxy = null
+        return null
         break
     }
+    SafeRetry<GenericJson> proxyRetry = new SafeRetry<GenericJson>()
+    def retrievedTargetProxy = proxyRetry.doRetry(
```

Comment:
Sounds like a good idea. I think I'll do that in another PR to keep changes managable.